pyrometheus/chem_expr.py

```diff
@@ -159,15 +159,16 @@ def gen(c, t):
 # {{{ Data-handling helper
 
 def _zeros_like(argument):
-    # FIXME: This mishandles NaNs.
+    # Fixme: This mishandles NaNs.
     return 0 * argument
 
 # }}}
 
 # {{{ Equilibrium constants
 
 
-def equilibrium_constants_expr(sol: ct.Solution, react: ct.Reaction, gibbs_rt):
+def equilibrium_constants_expr(sol: ct.Solution, react: ct.Reaction,
+                               reaction_index, gibbs_rt):
```

Comment:
The arguments to this seem redundant now, and redundancy breeds inconsistency. Drop the `react` parameter?

Reply:
We still need `react` because we still get values out of it, such as the indices of its reactants and products:

[https://github.com/pyrometheus/pyrometheus/blob/437f5663b346283ba81cee51421b5cf274eb1998/pyrometheus/chem_expr.py#L178](`indices_reac = [sol.species_index(sp) for sp in react.reactants]`)

The problem is that, to the best of my knowledge, a Cantera-2.6 `Reaction` does not know the reaction's "global" index in a mechanism (in Cantera 2.5, it had `ID`).

Reply:
> The problem is that, to the best of my knowledge, a Cantera-2.6 Reaction does not know the reaction's "global" index in a mechanism (in Cantera 2.5, it had ID).

But it looks like you'll be able to get the reaction given the `sol` and the index? (Or am I missing something?)

Reply:
Indeed, that's sounds like the right way to do it... I hadn't had coffee when I looked at this this morning.